client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceTool.js

```diff
@@ -0,0 +1,434 @@
+      } = deploymentConfig;
+
+      const processInstance =
+        await this.startWithConfiguration(startConfiguration, processDefinition, endpoint);
+
+      await this.handleStartSuccess(processInstance, endpoint);
+    } catch (error) {
+      displayNotification({
+        type: 'error',
+        title: START_INSTANCE_FAILED,
+        content: 'See the log for further details.',
+        duration: 10000
+      });
+      log({ category: 'start-instance-error', message: error.problems || error.message });
+    }
+  }
+
+  async saveConfiguration(tab, configuration) {
+    const {
+      config
+    } = this.props;
+
+    await config.setForFile(tab.file, START_DETAILS_CONFIG_KEY, configuration);
+
+    return configuration;
+  }
+
+  canStartWithConfiguration(configuration) {
+
+    if (!configuration) {
+      return false;
+    }
+
+    const {
+      businessKey
+    } = configuration;
+
+    return !!businessKey;
+  }
+
+  async getDefaultConfiguration(providedConfiguration = {}) {
+
+    const startInstance = providedConfiguration || {};
+
+    return {
+      businessKey: 'default',
+      ...startInstance
+    };
+  }
+
+  async getSavedConfiguration(tab) {
+    const {
+      config
+    } = this.props;
+
+    return config.getForFile(tab.file, START_DETAILS_CONFIG_KEY);
+  }
+
+  async getSavedProcessDefinition(tab) {
+    const {
+      config
+    } = this.props;
+
+    return config.getForFile(tab.file, PROCESS_DEFINITION_CONFIG_KEY);
+  }
+
+  async saveProcessDefinition(tab, deployment) {
+    if (!deployment || !deployment.deployedProcessDefinition) {
+      return;
+    }
+
+    const {
+      deployedProcessDefinition: processDefinition
+    } = deployment;
+
+    const {
+      config
+    } = this.props;
+
+    return await config.setForFile(tab.file, PROCESS_DEFINITION_CONFIG_KEY, processDefinition);
+  }
+
+  async getConfigurationFromUserInput(tab, providedConfiguration, uiOptions) {
+    const configuration = await this.getDefaultConfiguration(providedConfiguration);
+
+    return new Promise(resolve => {
+      const handleClose = (action, configuration) => {
+
+        this.setState({
+          modalState: null
+        });
+
+        // contract: if configuration provided, user closed with O.K.
+        // otherwise they canceled it
+        return resolve({ action, configuration });
+      };
+
+      this.setState({
+        modalState: {
+          tab,
+          configuration,
+          handleClose,
+          ...uiOptions
+        }
+      });
+    });
+  }
+
+  startWithConfiguration(configuration, processDefinition, endpoint) {
+
+    const api = new CamundaAPI(endpoint);
+
+    return api.startInstance(processDefinition, configuration);
+  }
+
+  handleStartSuccess(processInstance, endpoint) {
+    const {
+      displayNotification
+    } = this.props;
+
+    const {
+      url
+    } = endpoint;
+
+    displayNotification({
+      type: 'error',
+      title: 'Process instance started successfully',
```

Comment:
Not sure whether successful process instance start should be of type `error` 🤔

Reply:
😄

Reply:
Updated it.